File: aoc/01/solution.py
import itertools


def part_one(data):
    data = [int(n) for n in data.splitlines()]

    for x, y in itertools.combinations(data, r=2):
        if x + y == 2020:
            print(x * y)
            break


def part_two(data):
    data = [int(n) for n in data.splitlines()]

    for x, y, z in itertools.combinations(data, r=3):
        if x + y + z== 2020:
            print(x * y * z)
            break


if __name__ == '__main__':
    with open('aoc/01/input.txt') as f:
        problem_input = f.read()

    part_one(problem_input)
    part_two(problem_input)

File: aoc/01/solution.rs
fn main() {
    let data = get_input();

    println!("Day 01 Part 01: {}", part_one(&data).unwrap());
    println!("Day 01 Part 02: {}", part_two(&data).unwrap());
}

fn get_input() -> Vec<i64> {
    let input = String::from(include_str!("input.txt"));
    
    input
        .lines()
        .map(|num| num.parse::<i64>().unwrap())
        .collect()
}

fn part_one(data: &Vec<i64>) -> Option<i64> {
    for x in data {
        let y = 2020 - x;

        if data.contains(&y) {
            return Some(x * y);
        }
    }
    None
}

fn part_two(data: &Vec<i64>) -> Option<i64> {
    for x in data {
        for y in data {
            let z = 2020 - y - x;

            if data.contains(&z) {
                return Some(x * y * z);
            }
        }
    }
    None
}
